src/main/java/duke/DukeParser.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.command.Command;
+import duke.command.CommandAddDeadline;
+import duke.command.CommandAddEvent;
+import duke.command.CommandAddTodo;
+import duke.command.CommandDelete;
+import duke.command.CommandDone;
+import duke.command.CommandExit;
+import duke.command.CommandHelp;
+import duke.command.CommandInvalid;
+import duke.command.CommandList;
+import task.TaskList;
+
+/**
+ * Parses any command that requires inputs
+ */
+public class DukeParser {
+
+    private final TaskList taskList;
+
+    /**
+     * Patterns for the Parser to look out for in the input
+     */
+    private final Pattern listPattern = Pattern.compile("list( .+)?", Pattern.CASE_INSENSITIVE);
+    private final Pattern donePattern = Pattern.compile("done (\\d+)", Pattern.CASE_INSENSITIVE);
+    private final Pattern deletePattern = Pattern.compile("delete (\\d+)", Pattern.CASE_INSENSITIVE);
+    private final Pattern todoPattern = Pattern.compile("todo (.+)", Pattern.CASE_INSENSITIVE);
+    private final Pattern deadlinePattern = Pattern.compile(
+            "deadline (.+) /by (\\d{1,2}/\\d{1,2}/\\d{4}+)( \\d{4}+)?",
+            Pattern.CASE_INSENSITIVE);
+    private final Pattern eventPattern = Pattern.compile(
+            "event (.+) /at (\\d{1,2}/\\d{1,2}/\\d{4}+)( \\d{4}+)?",
+            Pattern.CASE_INSENSITIVE);
+
+    /**
+     * Constructor; instantiates the task list to be edited
```

Comment:
First word after semi-colon should be capitalized.

Reply:
Ah totally missed this out! Thanks